web/src/server/utils/checkClickhouseAccess.ts

```diff
@@ -27,125 +20,6 @@ export const measureAndReturnApi = async <T, Y>(args: {
-      }
-
-      // logic for regular users:
-      // if env.LANGFUSE_READ_FROM_POSTGRES_ONLY is true, return postgres only
-      // otherwise fetch both and compare timing
-      // if env.LANGFUSE_RETURN_FROM_CLICKHOUSE is true, return clickhouse data
-
-      const isExcludedFromClickhouse =
-        user?.featureFlags.excludeClickhouseRead ?? false;
-
-      const excludedProjects =
-        env.LANGFUSE_EXPERIMENT_EXCLUDED_PROJECT_IDS?.split(",") ?? [];
-
-      const isExcludedProject = excludedProjects.includes(
-        input.projectId ?? "",
-      );
-
-      if (
-        env.LANGFUSE_READ_FROM_POSTGRES_ONLY === "true" ||
-        isExcludedFromClickhouse ||
-        isExcludedProject
-      ) {
-        logger.info("Read from postgres only");
-        return await pgExecution(input);
-      }
-
-      logger.debug("Read from postgres and clickhouse");
-      try {
-        const [[pgResult, pgDuration], [chResult, chDuration]] =
-          await Promise.all([
-            executionWrapper(input, pgExecution, currentSpan, "pg"),
-            executionWrapper(input, clickhouseExecution, currentSpan, "ch"),
-          ]);
-        // Positive duration difference means clickhouse is faster
-        const durationDifference = pgDuration - chDuration;
-        currentSpan?.setAttribute(
-          "execution-time-difference",
-          durationDifference,
-        );
-        currentSpan?.setAttribute("pg-duration", pgDuration);
-        currentSpan?.setAttribute("ch-duration", chDuration);
-
-        recordHistogram("langfuse.clickhouse_experiment", chDuration, {
-          operation: args.operation,
-          database: "clickhouse",
-        });
-        recordHistogram("langfuse.clickhouse_experiment", pgDuration, {
-          operation: args.operation,
-          database: "postgres",
-        });
-
-        // if (args.operation.includes("dataset")) {
-        //   logger.info(
-        //     `operation: ${args.operation} pg result: ${JSON.stringify(pgResult)}, ch result: ${JSON.stringify(chResult)}`,
-        //   );
-        //   return pgResult;
-        // }
-
-        return env.LANGFUSE_RETURN_FROM_CLICKHOUSE === "true"
-          ? chResult
-          : pgResult;
-      } catch (e) {
-        logger.error(
-          "Error in clickhouse experiment wrapper. Retrying leading store.",
-          e,
-        );
-        return env.LANGFUSE_RETURN_FROM_CLICKHOUSE === "true"
-          ? clickhouseExecution(input)
-          : pgExecution(input);
-      }
-    },
-  );
-};
-
-const executionWrapper = async <T, Y>(
-  input: T & { queryClickhouse: boolean },
-  fn: (input: T) => Promise<Y>,
-  span?: opentelemetry.Span,
-  attributePrefix?: string,
-): Promise<[Y, number]> => {
-  const startTime = Date.now();
-  const res = await fn(input);
-  const duration = Date.now() - startTime;
-  span?.setAttribute(`${attributePrefix}-duration`, duration);
-  return [res, duration];
+  const { input, clickhouseExecution } = args;
```

Comment:
I'll clean this up in a second step, but want to keep the size of this somewhat manageable.